CIP-0118/README.md

```diff
@@ -0,0 +1,1041 @@
+
+This is a simple extension of the previous example.
+Instead of Party A, we have a set of parties A1 ... An who want to make various kinds of swap, and a batcher, Party B, who collects these and resolves them using some source of liquidity (in this example a big UTXO).
+
+1. Parties A1 ... An create transactions T1 ... Tn with requests and offers (unlocked outputs) representing their desired trade, as in the previous example.
+2. Parties A1 ... An send T1 ... Tn to B
+3. Party B creates two transactions:
+    - T(n+1) has fulfilling all the requests in T1 ... Tn, and spends Party B’s liquidity UTXO `u` in order to do so, creating a new entry `u'` with the remainder. It also requires T(n+2)
+    - T(n+2) spends the offers from T1 ... Tn as well as `u'`, creating a new increased liquidity output `u''`
+4. Party B submits a validation zone consisting of T1 ... T(n+2)
+
+
+### Babel fees
+
+Babel fees is a specific subtype of the first usecase where the request is necessarily
+a quantity of Ada. Usually, it also implies that no Ada is contained in the inputs
+of the request-making transaction, and the requested Ada is used to pay transaction fees.
+
+
+### DApp fee and min-UTxO sponsorship
+
+Party A wants to use a dApp operated by Party B (specifically, submit a transaction using a script S associated with the dApp).
+Party B wants to cover the fees and min-UTxO value for this.
+
+1. Party A creates a transaction T1 that uses script S, and has request `r` which requires Ada to cover the script fees.
+2. Party A sends T1 to Party B
+3. Party B creates a transaction T2 that fulfills `r`
+4. Party B submits a validation zone consisting of T1 and T2
+
+Note here that no offers are present in T1 because the the point of Dapp sponsorship
+is for the Dapp to cover the fees and min-UTxO value without being compensated in
+other tokens (although the Dapp itself may require depositing assets into it).
+
+### Bridges
+
+Centralized approaches to bridging tokens across blockchains have been developed,
+e.g. by [SingularityNET](https://singularitynet.gitbook.io/welcome-to-singularitynet/bridge/overview).
+A decentralized approach is much more difficult due to the requirement that
+there must be some way to verify data about the operation of an external piece of
+distributed software (on the side of both chains involved).
+This CIP makes the centralized version of the bridging process slightly more
+streamlined on the side of Cardano. Let B be a _trusted_ bridge. The untrusted
+case requires additional research.
+
+1. Party A sends an transaction T1 disposing of some tokens on chain D (either by burning
+  them or putting them in a special token pool) to B as well as to chain D
+2. Party A sends a transaction T2 with a request for the same amount of
+  tokens as in (1) on Cardano to B
+3. Bridge B waits for T1 to settle on D, then submits the zone `[T2; T3]` to
+  Cardano, where `T3` fulfills the request of `T2`
+
+
+
+## Towards Better Designs
```

Comment:
```suggestion
### Towards Better Designs
```
(with header level changes all the way down from here, and maybe part of the way up)

@polinavino this material is important & exactly in keeping with our goals for _Rationale_... to compare & contrast with other potential designs.  To put it in an unconventional (and therefore unexpected) top level section would likely be a distraction to reviewers and potential implementors.  For readers it does make sense to keep it in a separate section, but the section title & order it appears in the source material is sufficient here to categorise it as future rather than current consideration without making it a top-level heading.